google/cloud/storage/internal/grpc_split_write_object_data.cc

```diff
@@ -21,20 +21,7 @@ namespace cloud {
 namespace storage_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
-SplitObjectWriteData<std::string>::SplitObjectWriteData(
-    absl::string_view buffer)
-    : buffers_({buffer}),
-      total_size_(storage::internal::TotalBytes(buffers_)) {}
-
-SplitObjectWriteData<std::string>::SplitObjectWriteData(
-    google::cloud::storage::internal::ConstBufferSequence buffers)
-    : buffers_(std::move(buffers)),
-      total_size_(storage::internal::TotalBytes(buffers_)) {}
-
-bool SplitObjectWriteData<std::string>::Done() const {
-  return offset_ >= total_size_;
-}
-
+template <>
 std::string SplitObjectWriteData<std::string>::Next() {
   auto constexpr kMax = static_cast<std::size_t>(
       google::storage::v2::ServiceConstants::MAX_WRITE_CHUNK_BYTES);
```

Comment:
Aside: It looks like we can return more that `MAX_WRITE_CHUNK_BYTES`, which sounds odd.

Reply:
How?  `result` is always grown by at most `kMax - result.size()` bytes:

```
auto n = (std::min)(kMax - result.size(), b.size() - start);
```

and as soon as it reaches `kMax` the function returns.  I guess I should write `if (result.size() == kMax)`?

Reply:
My bad.  Sorry.  But, yes, `if (result.size() == kMax)` would make that clearer.